be/src/exec/olap_common.h

```diff
@@ -537,9 +537,17 @@ inline void ColumnValueRange<T>::convert_to_fixed_value() {
         }
         _fixed_op = FILTER_IN;
     } else {
-        for (T v = _low_value; v <= _high_value; helper::increase(v)) {
-            _fixed_values.insert(v);
+        // if _low_value == _high_value == type::max
+        // v will overflow after increase, so we have to
+        // do some special treatment
+        if (_low_value == _high_value) {
+            _fixed_values.insert(_high_value);
+        } else {
+            for (T v = _low_value; v <= _high_value; helper::increase(v)) {
```

Comment:
if v = INT_MAX,  after  helper::increate(v), it would be INT_MIN, the condition v<=_high_value is hold. so in case of _high_value == INT_MAX, it's better to process this case standalonely.